test/utils/types/Time.test.js

```diff
@@ -0,0 +1,150 @@
+require('@openzeppelin/test-helpers');
+
+const { expect } = require('chai');
+const { clock } = require('../../helpers/time');
+
+const Time = artifacts.require('$Time');
+
+const unpackDelay = delay => ({
+  valueBefore: (BigInt(delay) >> 32n) % (1n << 32n),
+  valueAfter: (BigInt(delay) >> 0n) % (1n << 32n),
+  effect: (BigInt(delay) >> 64n) % (1n << 48n),
+});
+
+const packDelay = ({ valueBefore, valueAfter = 0n, effect = 0n }) =>
+  (BigInt(valueAfter) % (1n << 32n) << 0n) +
+  (BigInt(valueBefore) % (1n << 32n) << 32n) +
+  (BigInt(effect) % (1n << 48n) << 64n);
+
+const max = (first, ...values) => values.reduce((x, y) => (x > y ? x : y), first);
+
+const MAX_UINT32 = 1n << (32n - 1n);
+const MAX_UINT48 = 1n << (48n - 1n);
+
+contract('Time', function () {
+  beforeEach(async function () {
+    this.mock = await Time.new();
+  });
+
+  describe('clocks', function () {
+    it('timestamp', async function () {
+      expect(await this.mock.$timestamp()).to.be.bignumber.equal(web3.utils.toBN(await clock.timestamp()));
+    });
+
+    it('block number', async function () {
+      expect(await this.mock.$blockNumber()).to.be.bignumber.equal(web3.utils.toBN(await clock.blocknumber()));
+    });
+  });
+
+  it('isSetAndPast', async function () {
+    for (const timepoint of [0n, 1n, 2n, 17n, 42n, MAX_UINT48])
+      for (const ref of [0n, 1n, 2n, 17n, 42n, MAX_UINT48]) {
+        expect(await this.mock.$isSetAndPast(timepoint, ref)).to.be.equal(timepoint != 0 && timepoint <= ref);
+      }
+  });
+
+  describe('Delay', function () {
+    describe('helpers', function () {
```

Comment:
Is the goal of this section to test the JS helpers or the Solidity code?

```suggestion
    describe('packing and unpacking', function () {
```